### ISSUE TITLE
Incorrect error messages in health_check for SSL configuration issues in Manager 3.2.8

### ISSUE THREAD

**Issue by mikliapko** [scylladb/scylla-manager#3889]
**Test description:**
The test starts a (second) cluster with ssl disabled, and adds it to the manager.

Afterwards, the test enables ssl encryption for the cluster, without updating the manager, and because of that the manager cannot communicate with the cluster through cql.

At the end, the test requests the status of the cluster from the manager, and makes sure that the cql status of all of the nodes is ERROR, and that proper error messages were printed for each of the nodes, since the manager ('s agents) fail to communicate with the cluster due to the missing ssl keys.

**Actual result:**
Manager 3.2.8 returns incorrect error messages:
```
- 127.0.71.1 alternator: get node info: no host config available
- 127.0.71.1 CQL: no host config available
```

**Expected result:**
The manager should return error messages indicating that SSL client encryption is enabled but the certificate is missing. 
Example from version 3.2.7:
```
- 127.0.70.1 alternator: get node info: client encryption is enabled, but certificate is missing: get SSL user cert from secrets store: not found
- 127.0.70.1 CQL: client encryption is enabled, but certificate is missing: get SSL user cert from secrets store: not found
```

**Environment:**
Scylla manager - 3.2.8

**Additional Info:**
- The issue does not occur in Manager 3.2.7, where the correct and informative error messages are returned.
- Link to manager [logs](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/manager-dtest/1582/artifact/scylla-dtest/logs/1718838591803_manager_health_check_tests.py%3A%3ATestManagerHealthCheck%3A%3Atest_error_cql_status_when_ssl_is_activated/scylla-manager.log)
- CI [job](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/manager-dtest/1582)

**Comment by gdubicki** [comment]
We are having this problem after an upgrade to Scylla Manager 3.2.8 in production.

Should we just downgrade to 3.2.7 until it's resolved? Or is it fixed in 3.3.0 and we should upgrade?

**Comment by karol-kokoszka** [comment]
@gdubicki 
```
- 10.7.241.130 alternator: get node info: no host config available
- 10.7.241.130 CQL: no host config available
```
These errors comes from the config cache service introduced with manager 3.2.8. and they indicate that service couldn't update cache with the latest / correct scylla node configuration, as some of the information is missing in manager DB.

```
    "error": "retrieve cluster host configuration: building node config: unable to create TLS configuration for CQL session: client encryption is enabled, but certificate is missing: not found",
```
Means that even though the `scylla.yaml` enables scylla encryption, and requires the client authentication, the certificates are not provided to manager. Due to that, it cannot establish secured CQL session.

Check your `scylla.yaml` files on nodes against:
```
client_encryption_options:
  enabled: true     <------ this seems to be enabled
  certificate: /etc/scylla/db.crt     <- missing in manager db
  keyfile: /etc/scylla/db.key   <- missing in manager db
#    truststore: <none, use system trust>
  require_client_auth: true    <---- this seems to be enabled
#    priority_string: <not set, use default>
```

Check https://manager.docs.scylladb.com/stable/sctool/cluster.html#ssl-user-cert-file https://manager.docs.scylladb.com/stable/sctool/cluster.html#ssl-user-key-file

If you want to disable TLS, use this flag https://manager.docs.scylladb.com/stable/sctool/cluster.html#force-tls-disabled , but pls make sure that non-TLS CQL session is allowed in your scylla configuration.

**Comment by gdubicki** [comment]
Thanks @karol-kokoszka!

But the thing is that our `scylla.yaml` has only this config:
```
read_request_timeout_in_ms: 5000
write_request_timeout_in_ms: 2000
cas_contention_timeout_in_ms: 1000

consistent_cluster_management: true
```
...and if I am reading https://github.com/scylladb/scylladb/blob/scylla-5.4.7/conf/scylla.yaml#L474 right, the default setting is disabled. 😕

**Comment by karol-kokoszka** [comment]
@gdubicki for some reason the node in you cluster reported on some nodes (or on all nodes) the encryption enabled.
Here is the SM part building the cached node configuration:
https://github.com/scylladb/scylla-manager/blob/8d9190b5a0e12e0ec0ef611aa9295e703b30d741/pkg/service/configcache/tlsconfig.go#L22-L47

Here is the API call to scylla-server checking the encryption options:
https://github.com/scylladb/scylla-manager/blob/8d9190b5a0e12e0ec0ef611aa9295e703b30d741/swagger/gen/scylla/v2/client/config/config_client.go#L1153-L1181

You can jump to some of the cluster node and call 
```
curl 127.0.0.1:10000/v2/config/client_encryption_options
```
to see what scylla-server reports.

**Comment by gdubicki** [comment]
Formatted response to `curl 127.0.0.1:10000/v2/config/client_encryption_options`:
```
{
    "truststore": "/var/run/secrets/scylla-operator.scylladb.com/scylladb/client-ca/tls.crt",
    "optional": "false",
    "keyfile": "/var/run/secrets/scylla-operator.scylladb.com/scylladb/serving-certs/tls.key",
    "enabled": "true",
    "require_client_auth": "true",
    "certificate": "/var/run/secrets/scylla-operator.scylladb.com/scylladb/serving-certs/tls.crt"
}
```

So I updated our `scylla.yaml` to include a - safer in my opinion - setting to make encryption optional and it resolved my problem! Thanks @karol-kokoszka! <3

**Comment by gdubicki** [comment]
Sorry bother you again, @karol-kokoszka, but I can't find a way to do a similar change for the Scylla Manager cluster... It seems to only have a generated ConfigMap named `scylladb-managed-config`:

```
apiVersion: v1
data:
  scylladb-managed-config.yaml: |
    cluster_name: "scylla"
    rpc_address: "0.0.0.0"
    endpoint_snitch: "GossipingPropertyFileSnitch"
    internode_compression: "all"
    native_transport_port_ssl: 9142
    native_shard_aware_transport_port_ssl: 19142
    client_encryption_options:
      enabled: true
      optional: false
      certificate: "/var/run/secrets/scylla-operator.scylladb.com/scylladb/serving-certs/tls.crt"
      keyfile: "/var/run/secrets/scylla-operator.scylladb.com/scylladb/serving-certs/tls.key"
      require_client_auth: true
      truststore: "/var/run/secrets/scylla-operator.scylladb.com/scylladb/client-ca/tls.crt"
kind: ConfigMap
metadata:
  annotations:
    meta.helm.sh/release-name: scylla
    meta.helm.sh/release-namespace: scylla
    scylla-operator.scylladb.com/managed-hash: <redacted>
==
  creationTimestamp: "<redacted>"
  labels:
    app.kubernetes.io/managed-by: Helm
    scylla/cluster: scylla
  name: scylla-managed-config
  namespace: scylla
  ownerReferences:
  - apiVersion: scylla.scylladb.com/v1
    blockOwnerDeletion: true
    controller: true
    kind: ScyllaCluster
    name: scylla
    uid: <redacted>
  resourceVersion: "<redacted>"
  uid: <redacted>
```

I can't find anything about modifying it in the https://operator.docs.scylladb.com/stable/helm.html...

**Comment by karol-kokoszka** [comment]
@gdubicki Looks that you are using operator. Please raise the ticket in the scylla-operator https://github.com/scylladb/scylla-operator

@tnozicka are you able to transfer issue from here to operator repo ?

**Comment by tnozicka** [comment]
> @tnozicka are you able to transfer issue from here to operator repo ?

nope, but give the issue started with something else then the config question, new issue seems appropriate + we have a template that should be followed

**Comment by Michal-Leszczynski** [comment]
@mikliapko do we still encounter this issue, or is it ok now?

**Comment by pehala** [comment]
We do encounter it. The dtest that tests it is just xfailed and the SSL is still not in the message. https://github.com/scylladb/scylla-dtest/blob/4db80f4c2d9ef21849b894f6bb5a7f1bfd2069f7/manager_health_check_tests.py#L211
